tests/functionnal/NotificationTargetTicket.php

```diff
@@ -139,5 +139,217 @@ public function testgetDataForObject() {
+         'users_id' => getItemByTypeName('User', 'tech', true),
+         'users_id_editor' => getItemByTypeName('User', 'tech', true),
+         'itemtype' => 'Ticket',
+         'is_private' => 0,
+         'items_id' => $tickets_id,
+      ]);
+      $this->integer($fup1_id)->isGreaterThan(0);
+
+      // Add followup from post_only
+      $fup_post_only = new \ITILFollowup();
+      $fup2_id = $fup_post_only->add([
+         'content' => 'test post_only',
+         'users_id' => getItemByTypeName('User', 'post-only', true),
+         'users_id_editor' => getItemByTypeName('User', 'post-only', true),
+         'itemtype' => 'Ticket',
+         'is_private' => 0,
+         'items_id' => $tickets_id,
+      ]);
+      $this->integer($fup2_id)->isGreaterThan(0);
+
+      // Add private followup to tech
+      $fup_private_tech = new \ITILFollowup();
+      $fup3_id = $fup_private_tech->add([
+         'content' => 'test private followup',
+         'users_id' => getItemByTypeName('User', 'tech', true),
+         'users_id_editor' => getItemByTypeName('User', 'tech', true),
+         'itemtype' => 'Ticket',
+         'is_private' => 1,
+         'items_id' => $tickets_id,
+      ]);
+      $this->integer($fup3_id)->isGreaterThan(0);
+
+      //add private task from tech
+      $task_private = new \TicketTask();
+      $task1_id = $task_private->add([
+         'state'             => \Planning::TODO,
+         'tickets_id'        => $tickets_id,
+         'tasktemplates_id'  => '0',
+         'is_private'        => 1,
+         'taskcategories_id' => '0',
+         'actiontime'        => "172800",                                  //1hours
+         'content'           => "Private Task",
+         'users_id_tech'     => getItemByTypeName('User', 'tech', true),
+      ]);
+      $this->integer($task1_id)->isGreaterThan(0);
+
+      //add task from tech
+      $task_tech = new \TicketTask();
+      $task2_id = $task_tech->add([
+         'state'             => \Planning::TODO,
+         'tickets_id'        => $tickets_id,
+         'tasktemplates_id'  => '0',
+         'taskcategories_id' => '0',
+         'is_private'        => 0,
+         'actiontime'        => "172800",                                  //1hours
+         'content'           => "Task",
+         'users_id_tech'     => getItemByTypeName('User', 'tech', true),
+      ]);
+      $this->integer($task2_id)->isGreaterThan(0);
+
+      // Add solution to test ticket
+      $solution = new \ITILSolution();
+      $solutions_id = $solution->add([
+         'content' => 'test',
+         'users_id' => getItemByTypeName('User', 'tech', true),
+         'users_id_editor' => getItemByTypeName('User', 'tech', true),
+         'itemtype' => 'Ticket',
+         'items_id' => $tickets_id,
+      ]);
+      $this->integer($solutions_id)->isGreaterThan(0);
+
+      $auth = new \Auth();
+      $this->boolean((boolean)$auth->login('tech', 'tech', true))->isTrue();
```

Comment:
Whye another login as `tech`? You should already be logged in (`$this->login('tech', 'tech');` at the beginning)

Reply:
indeed